### PR TITLE
Fixed delete to retain identifiers

### PR DESCRIPTION
**Pull request by nwrigh**
When we introduced the Identifier nodes we failed to also update the delete to cope with the mechanism.
We should be deleting all the nodes and the identifiers if there are no other relationships to other Things otherwise we remove all properties but retain the identifiers

### CODE REVIEW

File: people/people_service.go [hunk 1]
```diff
@@ -272,13 +270,15 @@ func (s service) Delete(uuid string) (bool, error) {
 		IncludeStats: true,
 	}
 
+	// Please note that this keeps the Identifiers as Identifiers are NOT a Thing
```
Comment (peteclark-ft) [inline people/people_service.go]: I'm not sure this does keep the identifiers? I read this query as: Given a Thing (with uuid), that has no relationships to other Things, delete the node, and all its identifiers.

File: people/people_service_test.go [hunk 2]
```diff
@@ -204,6 +218,31 @@ func TestDelete(t *testing.T) {
...
+
+	defer cleanDB([]string{fullPersonUuid, contentUUID}, db, t, assert)
+
+	assert.NoError(peopleDriver.Write(fullPerson), "Failed to write person")
+	writeContent(assert, db)
+	writeAnnotation(assert, db)
+
+	found, err := peopleDriver.Delete(fullPersonUuid)
+
+	assert.True(found, "Didn't manage to delete person for uuid %", fullPersonUuid)
```
Comment (peteclark-ft) [inline people/people_service_test.go:237]: `%s` ?

**Review by peteclark-ft — COMMENTED** [review]
Are we giving up on codecov.io reports as well?

File: people/people_service_test.go (hunk 2 above)
Comment (nwrigh) [reply people/people_service_test.go:237]: ooh yes where did that go :D

**Review by ghost — APPROVED** [review]

**Pull request merged**